includes/core-changes/corefx/5.0/assembly-api-behavior-changes-for-single-file-publish.md

```diff
@@ -0,0 +1,47 @@
+### Assembly-related API behavior changes for single-file publishing format
+
+Multiple APIs related to an assembly's file location have behavior changes when they're invoked in a single-file publishing format.
+
+#### Change description
+
+In single-file publishing for .NET 5.0 and later versions, bundled assemblies are loaded from memory instead of extracted to disk. For single-file published apps, this means that certain location-related APIs return different values on .NET 5.0 and later than on previous versions of .NET. The changes are as follows:
+
+| API | Previous versions | .NET 5.0 and later |
+| - | - | - |
+| <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> | Returns extracted DLL file path | Returns empty string for bundled assemblies |
+| <xref:System.Reflection.Assembly.CodeBase?displayProperty=nameWithType> | Returns extracted DLL file path | Throws exception for bundled assemblies |
+| <xref:System.Reflection.Assembly.GetFile(System.String)?displayProperty=nameWithType> | Returns `null` for bundled assemblies | Throws exception for bundled assemblies |
```

Comment:
Is the type of exception specified? Would that help modify any affected code?

Reply:
I think it's a NotImplementedException, but @agocke can you confirm?